src/context/WalletProvider/MetaMask/config.ts

```diff
@@ -1,13 +1,18 @@
+import type { MetaMaskAdapter } from '@shapeshiftoss/hdwallet-metamask'
 import { getConfig } from 'config'
 import { MetaMaskIcon } from 'components/Icons/MetaMaskIcon'
 import type { SupportedWalletInfo } from 'context/WalletProvider/config'
 
-export const MetaMaskConfig: Omit<SupportedWalletInfo, 'routes'> = {
+type MetaMaskConfigType = Omit<SupportedWalletInfo<typeof MetaMaskAdapter>, 'routes'>
+
+export const MetaMaskConfig: MetaMaskConfigType = {
   adapters: [
     {
       loadAdapter: () =>
         getConfig().REACT_APP_EXPERIMENTAL_MM_SNAPPY_FINGERS
-          ? import('@shapeshiftoss/hdwallet-shapeshift-multichain').then(m => m.MetaMaskAdapter)
+          ? import('@shapeshiftoss/hdwallet-shapeshift-multichain').then(
+              m => m.MetaMaskAdapter as typeof MetaMaskAdapter,
```

Comment:
is this cast still required?

Reply:
Still is unfortunately:
```
          Type 'Promise<MetaMaskShapeShiftMultiChainHDWallet | undefined>' is not assignable to type 'Promise<MetaMaskHDWallet | undefined>'.
            Type 'MetaMaskShapeShiftMultiChainHDWallet | undefined' is not assignable to type 'MetaMaskHDWallet | undefined'.
              Type 'MetaMaskShapeShiftMultiChainHDWallet' is missing the following properties from type 'MetaMaskHDWallet': _supportsSecretInfo, _supportsSecret, _supportsKava, _supportsKavaInfo, and 2 more. [2322]
```

This is the type error we're seeing, which is why we're lying to TS that this is always a `MetaMaskAdapter`